### PR TITLE
client/web: when readonly, add check for TS connection

### PR DESCRIPTION
**Pull request by soniaappasamy**
When the viewing user is accessing a webclient not over Tailscale, they must connect over Tailscale before being able to log into the full management client, which is served over TS. This change adds a check that the user is able to access the node's tailscale IP. If not able to, the signin button is disabled. We'll also be adding Copy here to help explain to the user that they must connect to Tailscale before proceeding.

Updates #10261

### CODE REVIEW

File: client/web/src/components/login-toggle.tsx [hunk 1]
```diff
@@ -81,6 +81,36 @@ function LoginPopoverContent({
...
+  /**
+   * canConnectOverTS indicates whether the current viewer
+   * is able to hit the node's web client that's being served
+   * at http://${node.IP}:5252. If false, this means that the
+   * viewer must connect to the correct tailnet before being
+   * able to sign in.
+   */
+  const [canConnectOverTS, setCanConnectOverTS] = useState<boolean>(false)
+
+  const checkTSConnection = useCallback(() => {
```
Comment (willnorris) [inline client/web/src/components/login-toggle.tsx:93]: I think we need an isLoading var or something here that's attached as a dependency to this callback. I was able to queue several dozen hanging connections by moving my mouse in and out of the component.  

We can also use an AbortController to set a shorter timeout, but I think the long timeout is fine as long as we restrict it to a single active connection at a time.

**Review by willnorris — APPROVED** [review]
seems to work in my testing.  Still need a bit of work on limiting multiple connections, and the TODO for the disabled state, but overall LGTM

File: client/web/src/components/login-toggle.tsx (hunk 1 above)
Comment (soniaappasamy) [reply client/web/src/components/login-toggle.tsx:93]: Oh the loading var is a good idea! I'll play with it tomorrow morning

Comment (soniaappasamy) [reply client/web/src/components/login-toggle.tsx:93]: Oh the loading var is a good idea! I'll play with it tomorrow morning

Comment (soniaappasamy) [reply client/web/src/components/login-toggle.tsx:94]: Added loading state, PTAL!

**Review by willnorris — APPROVED** [review]

**Pull request merged**